src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -859,8 +876,36 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
     # Setting the config dataplane endpoint
     config_dp_endpoint = get_config_dp_endpoint(cmd, connected_cluster.location)
 
+    if auto_upgrade != "false":
+        helm_values = get_all_helm_values(release_namespace, kube_config, kube_context)
+        auto_upgrade = helm_values.get('systemDefaultValues').get('azureArcAgents').get('autoUpdate')
+        if not auto_upgrade:
+            auto_upgrade = "false"
+
+    if default_registry:
+        container_registry_repository = None
+    else:
+        isCustomRegistryEnabled = helm_values.get('global').get('isCustomRegistryEnabled')
+        if isCustomRegistryEnabled is True and container_registry_repository is None:
+            container_registry_repository = helm_values.get('systemDefaultValues').get('image').get('repository')
+        if anonymous_pull is not None:
+            container_registry_username = ""
+            container_registry_password = ""
+        elif container_registry_username is None and container_registry_password is None:
+            container_registry_username = helm_values.get('systemDefaultValues').get('image').get('username')
+            container_registry_password = helm_values.get('systemDefaultValues').get('image').get('password')
+
     # Retrieving Helm chart OCI Artifact location
-    registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else utils.get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood, release_train_dogfood)
+    if container_registry_repository:
+        if auto_upgrade != "false":
+            auto_upgrade_error = 'Disable auto upgrade when using custom registry'
+            telemetry.set_exception(exception=auto_upgrade_error, fault_type=consts.Custom_Registry_Disable_Auto_Upgrade_Fault_Type, summary=auto_upgrade_error)
+            raise ArgumentUsageError(auto_upgrade_error, recommendation='use --auto_upgrade false')
+        registry_path = "{}/azurearck8s/azure-arc-k8sagents:{}".format(container_registry_repository, connected_cluster.agent_version)
+        # Send private registry information to telemetry
+        telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.CustomContainerRegistry': container_registry_repository})
+    else:
+        registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else utils.get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood, release_train_dogfood)
```

Comment:
So when a customer switches from pcr to mcr, we will be upgrading to latest agent version and not the pcr agent version?

Reply:
@shashankbarsin we are currently upgrading to latest version while moving from PCR to MCR. Do we need to stick to the PCR version or this is fine?

Reply:
We will stick to the PCR version. Will change the implementation

Reply:
> So when a customer switches from pcr to mcr, we will be upgrading to latest agent version and not the pcr agent version?

Whenever we do upgrade we check the current version of the cluster and pull chart of that particular version. This will retain the version and will not upgrade to the latest agent version